### PR TITLE
Introduce `ValidWaitParams`

### PR DESCRIPTION
**Pull request by abulenok**
Closes #

## Introduced changes

- ValidWaitParams::new ensures that wait_timeout and wait_retries are valid
- Created as a result of this discussion https://github.com/foundry-rs/starknet-foundry/pull/1710#discussion_r1490813951 

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/src/helpers/configuration.rs [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+use crate::ValidWaitParams;
 use anyhow::{anyhow, Result};
 use camino::Utf8PathBuf;
 use serde::{Deserialize, Serialize};
 use std::fs;
 use tempfile::TempDir;
 use toml::Value;
 
-#[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
+#[derive(Default, Deserialize, Serialize, Clone, Debug, PartialEq)]
```
Comment (THenry14) [inline crates/sncast/src/helpers/configuration.rs:10]: just to be sure you've checked - a default value for an Option is `None`, right? (I'm thinking about keystore param)

File: crates/sncast/src/lib.rs [hunk 2]
```diff
@@ -75,8 +76,54 @@ impl NumbersFormat {
 
 pub struct WaitForTx {
     pub wait: bool,
-    pub timeout: u16,
-    pub retry_interval: u8,
+    pub wait_params: ValidWaitParams,
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, Copy, PartialEq)]
+pub struct ValidWaitParams {
```
Comment (THenry14) [inline crates/sncast/src/lib.rs:83]: Let's just rename it to WaitParams, as it does not make sense to pass invalid params :)

**Review by THenry14 — APPROVED** [review]
Two comments that are nitpicks, so after addressing them this is mergeable :+1:

File: crates/sncast/src/helpers/configuration.rs (hunk 1 above)
Comment (abulenok) [reply crates/sncast/src/helpers/configuration.rs:10]: yeah

and `#[derive(Default)]` was suggested by clippy

File: crates/sncast/src/lib.rs (hunk 2 above)
Comment (abulenok) [reply crates/sncast/src/lib.rs:83]: maybe `ValidatedWaitParams`? :)

Comment (THenry14) [reply crates/sncast/src/lib.rs:83]: Still a bit redundant, but good enough for me :+1:

Comment (abulenok) [reply crates/sncast/src/lib.rs]: I just think that ValidatedWaitParams gives a signal that it's not just a struct collecting `retry_interval` and `timeout`, but a struct that also assures that these params are valid

**Pull request merged**